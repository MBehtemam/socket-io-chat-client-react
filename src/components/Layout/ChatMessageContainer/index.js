import styled from "styled-components";

const ChatMessageContainer = styled.div`
  display: flex;
  justify-content: ${props =>
    props.type === "send" ? "flex-end" : "flex-start"};
`;
export default ChatMessageContainer;
