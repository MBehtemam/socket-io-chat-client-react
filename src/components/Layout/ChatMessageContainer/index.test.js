import React from "react";
import styled from "styled-components";
import renderer from "react-test-renderer";
import "jest-styled-components";
import ChatMessageContainer from "./index";

describe("Test suits for ChatMessageContainer", () => {
  test("that must works", () => {
    const tree = renderer.create(<ChatMessageContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("that must have correct direction for send message", () => {
    const tree = renderer.create(<ChatMessageContainer type="send" />).toJSON();
    expect(tree).toHaveStyleRule("justify-content", "flex-end");
  });
  test("that must have correct direction for receive message", () => {
    const tree = renderer
      .create(<ChatMessageContainer type="receive" />)
      .toJSON();
    expect(tree).toHaveStyleRule("justify-content", "flex-start");
  });
});
