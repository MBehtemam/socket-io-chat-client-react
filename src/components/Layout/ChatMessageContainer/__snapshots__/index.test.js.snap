// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for ChatMessageContainer that must works 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  className="c0"
/>
`;
